app/src/main/java/org/dhis2/usescases/login/LoginPresenter.kt

```diff
@@ -92,6 +94,12 @@ class LoginPresenter(
         } ?: view.setUrl(view.getDefaultServerProtocol())
     }
 
+    fun trackServerVersion() {
+        val dhisVersion =
+            userManager?.d2?.systemInfoModule()?.systemInfo()?.blockingGet()?.version()!!
```

Comment:
Are we sure that this is not going to be null?. maybe, instead of !!, we can use:
`userManager?.d2?.systemInfoModule()?.systemInfo()?.blockingGet()?.version()?.let{ analyticsHelper.trackMatomoEvent(SERVER, VERSION, dhisVersion) }`

Reply:
sounds better 👍🏻  thanks I'll change it